### ISSUE TITLE
Allow using ts-node as the interpreter

### ISSUE THREAD

**Issue by felipecrs** [microsoft/vscode-test#92]
There is a problem that a lot of extensions suffer: when bundling an extension with webpack, we still depend on TypeScript to compile the test source code, and therefore we can't test the dist generated by webpack directly.

This brings the caveat of having to deal with multiple build tasks: `webpack --watch`, `tsc --watch` and properly switch them according to the environment we run. This pollutes both the scripts in package.json and also the `launch.json` and `tasks.json` in the `.vscode` folder.

If `vscode-test` was able to use `ts-node` to interpret the `.ts` files (such as `src/test/index.ts`), we would be able to circumvent this issue.

Mocha already supports doing such thing (`mocha --require ts-node/register`), so once `vscode-test` supports it we can tweak the `index.ts` to call `mocha` properly, so it also handles it.

If supported, we would be able to switch from:

```diff
// package.json
"scripts": {
    "vscode:prepublish": "npm run package",
    "compile": "webpack",
    "watch": "webpack --watch",
    "package": "webpack --mode production --devtool hidden-source-map",
-   "test-compile": "tsc -p ./",
-   "test-watch": "tsc -watch -p ./",
-   "pretest": "npm run test-compile && npm run lint",
+   "pretest": "npm run compile && npm run lint",
    "lint": "eslint src --ext ts",
-   "test": "node ./out/test/runTest.js"
+   "test": "ts-node ./src/test/runTest.ts"
},
```

```diff
// tasks.json 
{
	"version": "2.0.0",
	"tasks": [
		{
			"type": "npm",
			"script": "watch",
			"problemMatcher": [
                "$ts-webpack-watch",
                "$tslint-webpack-watch"
			],
			"isBackground": true,
			"presentation": {
				"reveal": "never"
			},
			"group": {
				"kind": "build",
				"isDefault": true
			}
		},
-		{
-			"type": "npm",
-			"script": "test-watch",
-			"problemMatcher": "$tsc-watch",
-			"isBackground": true,
-			"presentation": {
-				"reveal": "never"
-			},
-			"group": "build"
-		}
	]
}
```

```diff
// launch.json 
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Run Extension",
			"type": "extensionHost",
			"request": "launch",
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}",
				"${workspaceFolder}/src/test/fixture"
			],
			"outFiles": [
				"${workspaceFolder}/dist/**/*.js"
			],
			"preLaunchTask": "${defaultBuildTask}"
		},
		{
			"name": "Extension Tests",
			"type": "extensionHost",
			"request": "launch",
			"args": [
				"--extensionDevelopmentPath=${workspaceFolder}",
-				"--extensionTestsPath=${workspaceFolder}/out/test/index",
+				"--extensionTestsPath=${workspaceFolder}/src/test/index",
				"${workspaceFolder}/src/test/fixture"
			],
			"outFiles": [
-				"${workspaceFolder}/out/**/*.js"
+				"${workspaceFolder}/dist/**/*.js"
			],
-			"preLaunchTask": "npm: test-watch"
+			"preLaunchTask": "${defaultBuildTask}"
		}
	],
}
```

References https://github.com/microsoft/vscode-test/issues/19.

**Comment by vaclavHala** [comment]
I'd also very much like to see this. At the moment I managed to put together this hacky solution
```js
// entrypoint.js
function run() {
	const { compilerOptions } = require("./tsconfig.json");

	require("ts-node").register({compilerOptions});
	require("./mocha-entrypoint").runTests();
}
```
The bootstrap which assembles options and calls `runTests` I have written in pure js to avoid having to tsc the bootstrap file itself and the `entrypoint.js` gets passed as `--extensionTestsPath`. The only purpose of the `entrypoint,js` is to `register()` the `ts-node` loader with options from `tsconfig.json` in the tests folder while already inside vscode and then run the regular Mocha entrypoint. `./mocha-entrypoint.ts` and everything after it is typescript.

Big problem with this is that I can't get sourcemaps to work in vscode debug, no matter how I setup the `tsconfig.json` (I tried all combinations of `sourceMaps`, `inlineSourceMaps` and `inlineSources`). I know e.g. this https://github.com/webdriverio-community/wdio-vscode-service uses ts-node in such a way vscode can understand it and breakpoints in ts file work as expected, so I must be missing some crucial step/setting.

Could anyone more knowledgeable point me to some relevant docs on how exactly sourcemaps are loaded into vscode and how related problems can be debugger? I know about the `Debug diagnostics -> my breakpoints don't bind` but that does not tell me anything useful, only that source maps for my ts files can't be found
